src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -574,6 +574,35 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
                 "the os type of this image should be {}".format(community_gallery_image_info.os_type))
         namespace.os_type = community_gallery_image_info.os_type
 
+    if getattr(namespace, 'security_type', None):
+        if getattr(namespace, 'os_disk_security_encryption_type', None):
+            if namespace.security_type.lower() == 'confidentialvm' and \
+                    namespace.os_disk_security_encryption_type is None:
+                raise ArgumentUsageError('usage error: --os-disk-security-encryption-type is required'
+                                         ' when --security-type is specified as ConfidentialVM')
+            if namespace.os_disk_security_encryption_type != 'DiskWithVMGuestState' and \
+                    namespace.os_disk_secure_vm_disk_encryption_set is not None:
+                raise ArgumentUsageError(
+                    'usage error: The --os-disk-secure-vm-disk-encryption-set can only be passed in '
+                    'when --os-disk-security-encryption-type is DiskWithVMGuestState')
+        confidential_vm_sku = ['standard_dc2as_v5', 'standard_dc4as_v5', 'standard_dc8as_v5', 'standard_dc16as_v5',
+                               'standard_dc32as_v5', 'standard_dc48as_v5', 'standard_dc64as_v5', 'standard_dc96as_v5',
+                               'standard_dc2ads_v5', 'standard_dc4ads_v5', 'standard_dc8ads_v5', 'standard_dc16ads_v5',
+                               'standard_dc32ads_v5', 'standard_dc48ads_v5', 'standard_dc64ads_v5',
+                               'standard_dc96ads_v5', 'standard_ec2as_v5', 'standard_ec4as_v5', 'standard_ec8as_v5',
+                               'standard_ec16as_v5', 'standard_ec32as_v5', 'standard_ec48as_v5', 'standard_ec64as_v5',
+                               'standard_ec96as_v5', 'standard_ec2ads_v5', 'standard_ec4ads_v5', 'standard_ec8ads_v5',
+                               'standard_ec16ads_v5', 'standard_ec32ads_v5', 'standard_ec48ads_v5',
+                               'standard_ec64ads_v5', 'standard_ec96ads_v5',
+                               ]
+        if namespace.security_type.lower() == 'confidentialvm' and getattr(namespace, 'size', None):
+            if namespace.size.lower() not in confidential_vm_sku:
+                raise ArgumentUsageError('usage error: when --security-type is set to ConfidentialVM,'
+                                         ' the --size should be one of the following: {}'.format(confidential_vm_sku))
+        if namespace.security_type.lower() == 'confidentialvm' and getattr(namespace, 'vm_sku', None):
+            if namespace.vm_sku.lower() not in confidential_vm_sku:
+                raise ArgumentUsageError('usage error: when --security-type is set to ConfidentialVM,'
+                                         ' the --vm-sku should be one of the following: {}'.format(confidential_vm_sku))
```

Comment:
Since there are too many VM SKU here, we should not hard code them in the client code, otherwise they are likely to become obsolete and inconsistent with the service

Reply:
Because there are too many VM SKUs, I don't think the client should add this verification. Does service have relevant verification? @AjKundnani

![Screenshot 2022-06-29 205913](https://user-images.githubusercontent.com/16612065/176442070-61dec894-d4f0-4eee-8683-2c9041dcf9d3.png)

Reply:
> Because there are too many VM SKUs, I don't think the client should add this verification. Does service have relevant verification? @AjKundnani
> 
> ![Screenshot 2022-06-29 205913](https://user-images.githubusercontent.com/16612065/176442070-61dec894-d4f0-4eee-8683-2c9041dcf9d3.png)

@zhoxing-ms - Yes service has relevant verification available, client-side verification is not required as it will not be extensible, this was added to capture details as part of description.

Reply:
OK, thanks